### ISSUE TITLE
Help requested on Debugging non working flashing

### ISSUE THREAD

**Issue by samxplogs** [xitiomet/flipperzero-chat-esp32#1]
Hi Brian,
Thanks for you app but I am somhow struggling installing it :(

I was wondering if the following error/warnings could explain why it is not working at my side.
Flashing is OK but then I don't see any AP with the default SSID.

When doing ESP32 Sktech data Upload > FatFS I straight got
FatFS Error: partition size could not be found!

When Flashing I have this

flipperzero-chat-esp32-main\flipperzero-chat-esp32-main\flipperzero-chat-esp32\flipperzero-chat-esp32.ino: In function 'int registerUser(int, String&, String&, int)':
flipperzero-chat-esp32-main\flipperzero-chat-esp32-main\flipperzero-chat-esp32\flipperzero-chat-esp32.ino:181:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^

Thanks for your help

**Comment by xitiomet** [comment]
Hello samxplogs,

Sorry it took a few days for me to see this. I'd start by making sure you have the partition scheme set to Default with "FFat" 1.2/1.5 (image below)
![image](https://user-images.githubusercontent.com/120686/222276337-149b5498-8c41-4175-8381-f51be462ba68.png)

I also have noticed that Flash mode DIO seems to work best with PSRAM disabled.

As far as the source code warning goes, I see the problem, fixed on my end be sure to pull the latest source, i've made a few changes/bugfixes in the last couple days. including a new library requirement for an optional screen. (See updated README)

**Comment by samxplogs** [comment]
Thanks for the details, I have tried but still stuck. From what I understand the issue still come from the partition.
Because I have compiled with the settings.json inside the data folder.

Once compiled, it is stuck in bootloop with the following error. I have checked and ensure I have selectected the correct device in the board manager for my chip:ESP-WROOM-32 which should be ESP32 Dev Module.

Could you share with me the module you are using ?

11:43:00.646 -> ELF file SHA256: 0000000000000000
11:43:00.646 -> 
11:43:00.646 -> Rebooting...
11:43:00.693 -> ets Jun  8 2016 00:22:57
11:43:00.693 -> 
11:43:00.693 -> rst:0xc (SW_CPU_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
11:43:00.693 -> configsip: 0, SPIWP:0xee
11:43:00.693 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
11:43:00.693 -> mode:DIO, clock div:1
11:43:00.693 -> load:0x3fff0030,len:1100
11:43:00.693 -> ho 0 tail 12 room 4
11:43:00.693 -> load:0x40078000,len:12308
11:43:00.693 -> load:0x40080400,len:3076
11:43:00.693 -> entry 0x400805ec
11:43:01.023 -> INIT
**11:43:01.118 -> /settings.json doesn't exist**
11:43:01.213 -> assertion "Invalid mbox" failed: file "IDF/components/lwip/lwip/src/api/tcpip.c", line 455, function: tcpip_send_msg_wait_sem
11:43:01.213 -> 
11:43:01.213 -> abort() was called at PC 0x40164b93 on core 1
11:43:01.213 -> 
11:43:01.213 -> 
11:43:01.213 -> Backtrace:0x400f4195:0x3ffb25800x4008c905:0x3ffb25a0 0x4009270d:0x3ffb25c0 0x40164b93:0x3ffb2640 0x400febf3:0x3ffb2670 0x40113bb9:0x3ffb26a0 0x40113e58:0x3ffb26c0 0x400fe305:0x3ffb2710 0x400de37e:0x3ffb2730 0x400de3f8:0x3ffb2780 0x400e4907:0x3ffb27a0 0x400daa53:0x3ffb27c0 0x400ea977:0x3ffb2820 
11:43:01.213 ->

**Comment by xitiomet** [comment]
Ok it looks like the sketch successfully flashed, but not the FFat image..

Regarding the ESP32 from what i understand there should be no major difference between modules but just in case mine was built using these. https://www.amazon.com/dp/B09GK74F7N

After uploading the the code, did you get any errors trying to upload the filesystem?
![image](https://user-images.githubusercontent.com/120686/222501542-cbcf239f-51a4-4daf-b697-d31ce9de157c.png)

or did it successfully upload the FatFS image?

Should look something like this
[FatFS] data   : C:\Users\brian\src\flipperzero-chat-esp32\data
[FatFS] offset : 4096
[FatFS] start  : 2691072
[FatFS] size   : 1468
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\bootstrap.min.cssdir_d: /spiflash/bootstrap.min.css  file name: bootstrap.min.css
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\bootstrap.min.jsdir_d: /spiflash/bootstrap.min.js  file name: bootstrap.min.js
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\chat.cssdir_d: /spiflash/chat.css  file name: chat.css
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\flipper.pngdir_d: /spiflash/flipper.png  file name: flipper.png
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\index.htmldir_d: /spiflash/index.html  file name: index.html
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\index.jsdir_d: /spiflash/index.js  file name: index.js
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\jquery.min.jsdir_d: /spiflash/jquery.min.js  file name: jquery.min.js
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\reaper.pngdir_d: /spiflash/reaper.png  file name: reaper.png
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\send.pngdir_d: /spiflash/send.png  file name: send.png
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\settings.jsondir_d: /spiflash/settings.json  file name: settings.json
O>Pack file, dir_s: C:\Users\brian\src\flipperzero-chat-esp32\data\spy.pngdir_d: /spiflash/spy.png  file name: spy.png
O>g_flashmem[0]: �size: 1503232
[FatFS] upload : C:\Users\brian\AppData\Local\Temp\arduino_build_508725/flipperzero-chat-esp32.fatfs.bin
[FatFS] address: 2691072
[FatFS] port   : COM8
[FatFS] speed  : 921600
[FatFS] mode   : dio
[FatFS] freq   : 80m

**Comment by samxplogs** [comment]
Seems you spotted the issue here. I spent hours on it but I am still unable to install/make the mkfatfs tool works properly.
I have look at this doc https://github.com/lorol/arduino-esp32fs-plugin
I copied <home_dir>/Arduino/tools/ESP32FS/tool/esp32fs.jar and binary to C:\Users\Sam\AppData\Local\Arduino15\packages\esp32\tools

Tool Exit.
Chip : esp32
Using partition scheme from Arduino IDE.
Start: 0x291000
Size : 0x15f000

FatFS Error: mkfatfsnot found!

**Comment by samxplogs** [comment]
I have found this tutorial which helped me with the mkfatfs installation: https://www.mischianti.org/2021/04/06/esp32-integrated-ffat-fat-exfat-filesystem-6/

So No issues in Windows while flashing but issue remains the same.
ketch uses 908201 bytes (69%) of program storage space. Maximum is 1310720 bytes.
Global variables use 53224 bytes (16%) of dynamic memory, leaving 274456 bytes for local variables. Maximum is 327680 bytes.
esptool.py v4.5.1
Serial port COM3
Connecting....
Chip is ESP32-D0WDQ6 (revision v1.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 78:e3:6d:17:1d:44
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Flash will be erased from 0x00001000 to 0x00005fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000e000 to 0x0000ffff...
Flash will be erased from 0x00010000 to 0x000effff...
Compressed 17520 bytes to 12170...
Writing at 0x00001000... (100 %)
Wrote 17520 bytes (12170 compressed) at 0x00001000 in 0.5 seconds (effective 311.0 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 144...
Writing at 0x00008000... (100 %)
Wrote 3072 bytes (144 compressed) at 0x00008000 in 0.1 seconds (effective 227.8 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 47...
Writing at 0x0000e000... (100 %)
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.1 seconds (effective 493.1 kbit/s)...
Hash of data verified.
Compressed 913952 bytes to 589269...
Writing at 0x00010000... (2 %)
Writing at 0x0001aced... (5 %)
Writing at 0x00024715... (8 %)
Writing at 0x00036771... (11 %)
Writing at 0x0003c402... (13 %)
Writing at 0x0004205c... (16 %)
Writing at 0x00047c10... (19 %)
Writing at 0x0004d654... (22 %)
Writing at 0x00052f0a... (25 %)
Writing at 0x0005885e... (27 %)
Writing at 0x0005dc26... (30 %)
Writing at 0x000635df... (33 %)
Writing at 0x0006894d... (36 %)
Writing at 0x0006daee... (38 %)
Writing at 0x00072e63... (41 %)
Writing at 0x0007813d... (44 %)
Writing at 0x0007d326... (47 %)
Writing at 0x00082542... (50 %)
Writing at 0x00087979... (52 %)
Writing at 0x0008cdc8... (55 %)
Writing at 0x0009206c... (58 %)
Writing at 0x00097b76... (61 %)
Writing at 0x0009d290... (63 %)
Writing at 0x000a26b8... (66 %)
Writing at 0x000a7958... (69 %)
Writing at 0x000acdf1... (72 %)
Writing at 0x000b2404... (75 %)
Writing at 0x000b79f6... (77 %)
Writing at 0x000bd1b8... (80 %)
Writing at 0x000c2f2f... (83 %)
Writing at 0x000c886a... (86 %)
Writing at 0x000d0ed6... (88 %)
Writing at 0x000d9947... (91 %)
Writing at 0x000def28... (94 %)
Writing at 0x000e45c9... (97 %)
Writing at 0x000e9873... (100 %)
Wrote 913952 bytes (589269 compressed) at 0x00010000 in 10.8 seconds (effective 674.0 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...

When uploading the FatFS image, it is also OK but I still experience the same issue.

Chip : esp32
Using partition scheme from Arduino IDE.
Start: 0x291000
Size : 0x15f000
mkfatfs : C:\Users\Sam\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\tools\mkfatfs.exe

esptool : C:\Users\Sam\AppData\Local\Arduino15\packages\esp32\tools\esptool_py\4.5.1\esptool.exe

[FatFS] data   : C:\Users\Sam\Desktop\flipperzero-chat-esp32-main\flipperzero-chat-esp32\data
[FatFS] offset : 4096
[FatFS] start  : 2691072
[FatFS] size   : 1404
[FatFS] upload : C:\Users\Sam\AppData\Local\Temp\arduino_build_65300/flipperzero-chat-esp32.fatfs.bin
[FatFS] address: 2691072
[FatFS] port   : COM3
[FatFS] speed  : 921600
[FatFS] mode   : dio
[FatFS] freq   : 80m

->esptool.py v4.5.1
->Serial port COM3
->Connecting.........
->Chip is ESP32-D0WDQ6 (revision v1.0)
->Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
->Crystal is 40MHz
->MAC: 78:e3:6d:17:1d:44
->Uploading stub...
->Running stub...
->Stub running...
->Changing baud rate to 921600
->Changed.
->Configuring flash size...
->Auto-detected Flash size: 4MB
->Flash will be erased from 0x00291000 to 0x003effff...
->Compressed 1437696 bytes to 493496...
->Writing at 0x00291000... (3 %)
->Writing at 0x002af417... (6 %)
->Writing at 0x002c2814... (9 %)
->Writing at 0x002ce2b8... (12 %)
->Writing at 0x002da13b... (16 %)
->Writing at 0x002e5761... (19 %)
->Writing at 0x002e96fe... (22 %)
->Writing at 0x002ed6f9... (25 %)
->Writing at 0x002f16f4... (29 %)
->Writing at 0x002f56ad... (32 %)
->Writing at 0x002f96d3... (35 %)
->Writing at 0x002fd6f9... (38 %)
->Writing at 0x003016b8... (41 %)
->Writing at 0x003056bd... (45 %)
->Writing at 0x00309709... (48 %)
->Writing at 0x0030d704... (51 %)
->Writing at 0x003116ff... (54 %)
->Writing at 0x003156fa... (58 %)
->Writing at 0x003196be... (61 %)
->Writing at 0x0031d6e5... (64 %)
->Writing at 0x003216f7... (67 %)
->Writing at 0x00325734... (70 %)
->Writing at 0x0032972f... (74 %)
->Writing at 0x0032d72a... (77 %)
->Writing at 0x0033172a... (80 %)
->Writing at 0x00335725... (83 %)
->Writing at 0x00339720... (87 %)
->Writing at 0x0033d71b... (90 %)
->Writing at 0x00341716... (93 %)
->Writing at 0x00345738... (96 %)
->Writing at 0x0034b724... (100 %)
->Wrote 1437696 bytes (493496 compressed) at 0x00291000 in 11.8 seconds (effective 972.6 kbit/s)...
->Hash of data verified.
->
->Leaving...
->Hard resetting via RTS pin...

So flashing is OK
services are running 80 on 10.10.10.1 , but I am unable to reach the web server and neither the console. I will test again.
it finally worked after reflashing once again yes :)
